### PR TITLE
Add more Complex methods (part 2)

### PR DESCRIPTION
**Pull request by tekknolagi**
Based on #699 since that was accepted before I added more :)

Continues #608.

### CODE REVIEW

File: src/complex.rb [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+    return obj.numerator == 0
+  else
+    return obj == 0
+  end
+end
+
+def _exact_zero?(obj)
+  not imaginary.is_a?(Float) and _zero?(imaginary)
+end
+
```
Comment (seven1m) [inline src/complex.rb:16]: :thinking: I wonder if there's a way to have these little helper methods without polluting the `main` object...?

```
bin/natalie                                                                                                              3.0.4
nat> _zero?(0)
true
nat> methods.grep(/^_/)
[:_zero?, :_exact_zero?, :__send__, :__id__]
nat> 
```

Comment (seven1m) [reply src/complex.rb:16]: I guess It'd be better to pollute `Complex` than to pollute main...

Comment (tekknolagi) [reply src/complex.rb:16]: I could make it `private` on `Complex`. I wish there was a way to have module-private methods.

**Comment by seven1m** [comment]
Nice!

**Pull request merged**